packages/es-components/src/components/controls/buttons/DropdownButton.js

```diff
@@ -120,110 +120,101 @@ function arrowMovement(node) {
 
   node.addEventListener('keydown', handleArrowMovementKeys);
 
-  return () => {
+  return function removeArrowMovementListener() {
     node.removeEventListener('keydown', handleArrowMovementKeys);
   };
 }
 
-export class DropdownButton extends React.Component {
-  constructor(props) {
-    super();
+function DropdownButton(props) {
+  const [buttonValue, setButtonValue] = useState(props.buttonValue);
+  const [isOpen, setIsOpen] = useState(false);
 
-    this.state = {
-      buttonValue: props.buttonValue,
-      isOpen: false
-    };
-  }
-
-  componentDidMount() {
-    this.removeFocusTrapListener = focusTrap(this.buttonDropdown);
-    this.removeArrowMovementListener = arrowMovement(this.buttonDropdown);
-  }
-
-  componentDidUpdate(prevProps, prevState) {
-    if (prevState.isOpen !== this.state.isOpen) {
-      this.triggerButton.focus();
-    }
-  }
+  const buttonDropdown = React.createRef();
+  const triggerButton = React.createRef();
```

Comment:
I've been looking at the React source code. `createRef` only creates an object. `useRef` will memoize the value during the rendering phase. so, you can't set the value of `ref.current` during the rendering function when you use `createRef`, but you can when you invoke `useRef`.

this doesn't have any implications in this case because it's being used internally by React to set the current value to the given element. I don't think that we need to defer to one or the other at this point.